processor/src/main/java/org/mapstruct/ap/internal/model/BeanMappingMethod.java

```diff
@@ -126,11 +144,13 @@ public Builder returnTypeBuilder( BuilderType returnTypeBuilder ) {
 
         public Builder sourceMethod(SourceMethod sourceMethod) {
             this.method = sourceMethod;
+            super.method( sourceMethod );
```

Comment:
fix all referencing to the method to use the super version instead of managing it twice.